design/006-proposal-id-generation.md

```diff
@@ -0,0 +1,141 @@
+# Proposal: Mechanism for generating proposal ID
+
+* Author(s): Manoranjith, Matthias Geihs
+* Status: propose
+* Related issue: https://github.com/hyperledger-labs/go-perun/issues/167 
+
+## Summary
+
+This proposal describes the need for proposal ID and the mechanism for
+generating it.
+
+## Motivation
+
+Until now, Proposal ID was being generated by encoding all the parameters of a
+proposal using perunio encoding scheme. Recently, an abstraction for wire
+serialization was introduced (see
+[#233](https://github.com/hyperledger-labs/go-perun/issues/233)), so that
+different encoding schemes can be used for serializing wire messages. As a part
+of this abstraction, it was needed to the mechanism for generating proposal ID
+re-evaluate, because it depends on a specific encoding scheme: perunio
+encoding.
+
+Also, considering the plans to implement an alternate perun state channels
+client that can run on resource constrained embedded devices, documenting
+different parts of the perun protocol makes it easier to keep these two
+implementations consistent with each other.
+
+## Details
+
+### Need for proposal ID
+
+Fig 1 shows the sequence of interactions between two participants for opening a
+channel. It can be seen that, channel ID can be generated only after each of
+the participants know the nonce share of every other participant. Hence, until
+a channel ID can be generated, a unique identifier for identifying the channel
+proposals is needed. For this purpose, the proposal ID was introduced.
+
+### Expected properties for a proposal ID
+
+It is sufficient for the proposal ID to be unique for each channel proposal. It
+is not necessary to be able to derive the proposal ID from a channel proposal.
+
+### Currently used mechanism
+
+The proposal ID was generated by encoding the channel proposal along with the
+nonce share of proposer. Perun protocol allows multiple options for nonce
+share, such as using a monotonically increasing counters for each set of
+participants or filling a byte array using a cryptographically secure random
+number generator. In either case, the participant ensures that, the nonce share
+for each proposal between a given set of participants is different. Hence, the
+proposal ID was unique for each proposal generated by a participant.
+
+### Issues with currently used mechanism
+
+In [#233](https://github.com/hyperledger-labs/go-perun/issues/233),
+abstractions for wire serialization were introduced. After this, it is possible
+to use a different encoding schemes such as protocol buffers. For instance, if
```

Comment:
```suggestion
to use a different encoding scheme such as protocol buffers. For instance, if
```

Reply:
Updated in this [force-push](https://github.com/hyperledger-labs/perun-proposals/compare/5ac16903f87512ed316835ff43cc5b244448eb50..18b56f0b8584fd616aa1fcec4807b9d79c26767e).